src/app/RootLayout.tsx

```diff
@@ -23,23 +23,35 @@ import { Well } from 'react-bootstrap';
 import SheriffProfileModal from './containers/SheriffProfileModal';
 import ScheduleShiftCopyModal from './containers/ScheduleShiftCopyModal';
 import ScheduleShiftAddModal from './containers/ScheduleShiftAddModal';
+import { Glyphicon } from 'react-bootstrap';
+import { Collapse } from 'react-bootstrap';
+import Legend from './components/Legend/Legend';
 
-class Layout extends React.Component {
+class Layout extends React.Component<{}, { isLegendOpen: boolean, initialize?: boolean }> {
+  static defaultState = {
+    isLegendOpen: true
+  };
+
+  constructor(props: any, context: any) {
+    super(props, context);
+    this.state = { isLegendOpen: true };
+  }
 
   private onSelectCourthouse(id: string) {
     this.setState({ initialize: true });
   }
 
   render() {
     const needCourthouse = !(api as Client).isCourthouseSet;
+    const { isLegendOpen } = this.state;
```

Comment:
May want to wrap all of the state and display for this in a component instead of having it in the RootLayout